src/ssl.rs

```
@@ -0,0 +1,104 @@
+use crate::error::Result;
+use std::fmt;
+use std::io::Read;
+
+#[derive(Clone)]
+
+pub struct Ssl {
+    pub enabled: bool,
+    pub root_cert: Option<CertificateType>,
+}
+
+/*  Supports SSL for HTTPS connection
+///
+///     use prusto::ssl::{Ssl, CertificateType};
+///     use prusto::auth::Auth;
+///     use prusto::ClientBuilder;
+///   
+///     
+///     let cli = ClientBuilder::new("user", "host")
+///        .port(443)
+///        .auth(Auth::Basic("user".to_string(), Some("pwd".to_string())))
+///        .secure(true)
+///        .ssl(Ssl {
+///            enabled: true,
+///            root_cert: Some(CertificateType::Pem("/path/root.pem".to_string()))
+///        })
+///        .catalog("catalog")
+///        .build()
+///        .unwrap();
+*/
+impl Default for Ssl {
+    fn default() -> Self {
+        Ssl {
+            enabled: false,
+            root_cert: None,
+        }
+    }
+}
+
+#[derive(Clone)]
+pub enum CertificateType {
+    Pem(String),
+    Der(String),
+}
+
+impl CertificateType {
+    pub fn new_pem(root_certificate_path: &str) -> Result<reqwest::Certificate> {
+        let mut buf = Vec::new();
+        std::fs::File::open(root_certificate_path)
+            .expect(&format!(
+                "Error opening certificate file {}",
+                root_certificate_path
+            ))
```

I don't think `expect` is right here, we should return the error to the caller.